pkg/controller/cloud/pvlcontroller.go

```diff
@@ -205,9 +205,28 @@ func (pvlc *PersistentVolumeLabelController) createPatch(vol *v1.PersistentVolum
 	if newVolume.Labels == nil {
 		newVolume.Labels = make(map[string]string)
 	}
+	if len(volLabels) != 0 {
+		if newVolume.Spec.NodeAffinity == nil {
+			newVolume.Spec.NodeAffinity = new(v1.VolumeNodeAffinity)
+		}
+		if newVolume.Spec.NodeAffinity.Required == nil {
+			newVolume.Spec.NodeAffinity.Required = new(v1.NodeSelector)
+		}
+	}
+	requirements := make([]v1.NodeSelectorRequirement, len(volLabels))
+	i := 0
 	for k, v := range volLabels {
 		newVolume.Labels[k] = v
+		// Set NodeSelectorRequirements based on the labels
+		requirements[i] = v1.NodeSelectorRequirement{Key: k, Operator: v1.NodeSelectorOpIn, Values: []string{v}}
```

Comment:
So on GCE, we support Regional Disks, which are replicated in two zones.  To represent that, we actually stuffed both zones in a single label.

You'll have to treat the zone label specially.  And you'll need to use `volumeutil.LabelZonesToSet` in order to parse this label format.  And then, each zone should be included as a separate element in the Values.  See the example pv node affinity [here](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/volume-topology-scheduling.md#multi-zonal-volume)

Reply:
Drawing parallels with the `find*Labels()` methods, we could have something like `translate*Labels()` that implement plugin-specific translation logic like this.